tests/unit/test_bundle.py

```diff
@@ -1122,10 +1122,6 @@ async def __call__(self, *args, **kwargs):
 
         bundle = await handler._handle_local_charms(bundle, bundle_dir)
 
-        # TODO: for some reason 'assert_has_calls' is failing in
-        # Python3.5, refactor this with 'assert_has_calls' when
-        # Python3.5 support is dropped
-
```

Comment:
But has this refactor been done?

Reply:
No, because I think keeping the explicit checks helps to make it clear what is being tested and how. I'm not a huge fan making things concise in testing, it only conceals stuff and introduces more magic imho (... turns around and runs `tox` :facepalm:).

Here's the original change that switched to explicit https://github.com/juju/python-libjuju/commit/2dc598adbfb93d23279952f17a23a26cf3478459 , I think keeping it like this is better, but not a big deal really to change it back, let me know what you think.